shared/cache.go

```diff
@@ -194,15 +202,15 @@ func (mw *memcacheWriteCloser) Write(p []byte) (n int, err error) {
 
 func (mw *memcacheWriteCloser) Close() error {
 	mw.isClosed = true
-	if !mw.hasWritten {
+	if Clients.redisPool == nil || !mw.hasWritten {
```

Comment:
maybe return `errNoRedis` here

Reply:
I think it'd be a bit redundant with the same error on the reader side -- I don't think we currently have anywhere that writes directly to memcache without trying to read from it first; we'd end up having two "not connected to Redis" logs every request if Redis isn't available (e.g. when running locally). WDYT?

Reply:
Yea that's fine!